### PR TITLE
delete target on TargetAPI

### PR DESCRIPTION
**Pull request by smanini1**
#### Description:

Created deleteTarget func on TargetAPI
Remove targetAnnotation and targetOverlay to update homeController 

---

#### Preview:

ezgif com-video-to-gif (9)

### CODE REVIEW

File: ios-target/Networking/Services/TargetAPI.swift [hunk 1]
```diff
@@ -40,4 +41,18 @@ class TargetAPI {
...
   }
+  
+  class func deleteTarget(_ targetId: Int, success: (() -> Void)? = nil, failure: ((_ error: Error) -> Void)? = nil) {
+    let url = targetsUrl + String(targetId)
+    APIClient.request(.delete, url: url, success: {_, _ in 
+      success?()
+    }, failure: { error in
+      // Dirty fix for backend error
+      if error is Alamofire.AFError {
+        success?()
```
Comment (mcousillas6) [inline ios-target/Networking/Services/TargetAPI.swift:52]: why call success here? maybe if the error is not from the backend, you can default to something generic like `Something went wrong`

File: ios-target/Networking/Services/TargetAPI.swift [hunk 2]
```diff
@@ -40,4 +41,18 @@ class TargetAPI {
       failure(error)
     })
   }
+  
+  class func deleteTarget(_ targetId: Int, success: (() -> Void)? = nil, failure: ((_ error: Error) -> Void)? = nil) {
+    let url = targetsUrl + String(targetId)
+    APIClient.request(.delete, url: url, success: {_, _ in 
+      success?()
+    }, failure: { error in
+      // Dirty fix for backend error
```
Comment (pMalvasio) [inline ios-target/Networking/Services/TargetAPI.swift:50]: 🙈

**Review by mcousillas6 — COMMENTED** [review]
Nice! 👏

File: ios-target/View/Home/HomeViewController.swift [hunk 3]
```diff
@@ -78,6 +78,28 @@ class HomeViewController: UIViewController {
...
+        mapView.removeAnnotation(annotation)
+        removeTargetOverlay(targetId: targetId)
+      }
+    }
+  }
+  
+  func removeTargetOverlay(targetId: Int) {
+    guard
+      let overlays = mapView.overlays as? [TargetCircle],
+      let index = overlays.index(where: {$0.targetId == targetId})
```
Comment (pMalvasio) [inline ios-target/View/Home/HomeViewController.swift:99]: I know that using the `where` is more readable but maybe you can use it like: `overlays.index { $0.targetId == targetId }` Or at just add spaces before and after the brackets 🤷‍♂ Sorry, is my OCD talking 🙈

Comment (mcousillas6) [reply ios-target/View/Home/HomeViewController.swift:99]: Agree on the spaces 😬 . The trailing closure doesn't work on guard statements (don't ask me why, but the compiler 💥 )

**Review by pMalvasio — APPROVED** [review]

File: ios-target/Networking/Services/TargetAPI.swift (hunk 1 above)
Comment (smanini1) [reply ios-target/Networking/Services/TargetAPI.swift:52]: The problem is the backend should send a 204 which is a no content response but it sends a 200 so it detects an error ´Alamofire.AFError responseSerializationFailed´ 🤦🏽‍♀️. That's why it calls success when this is the error detected

**Review by mcousillas6 — APPROVED** [review]

File: ios-target/View/Home/HomeViewController.swift (hunk 3 above)
Comment (germanStabile) [reply ios-target/View/Home/HomeViewController.swift:99]: @smanini1 what swift version are you using? in Swift 5 this API is `firstIndex`

**Pull request merged**